core/wallet/transaction.go

```diff
@@ -531,6 +535,46 @@ func UpdatePendingTransactions(ctx context.Context, d ds.Datastore, cfg *config.
 	return scv1 + scv0, ecv1 + ecv0, nil
 }
 
+func CloseLedgerChannel(ctx context.Context, d ds.Datastore, conf *config.Config) error {
+	if err := Init(ctx, conf); err != nil {
+		return err
+	}
+	states, err := list(d)
+	if err != nil {
+		return err
+	}
+	if len(states) > 0 {
+		err = grpc.EscrowClient(escrowService).WithContext(ctx, func(ctx context.Context, c escrowPb.EscrowServiceClient) error {
+			now := time.Now()
+			for _, e := range states {
+				if now.Sub(e.TimeCreate) < 5*time.Minute {
+					continue
+				}
+				sig, err := Sign(e.State, hostWallet.privateKey)
+				if err != nil {
+					continue
+				}
+				_, err = c.CloseChannel(ctx, &ledgerPb.SignedChannelState{
+					Channel:     e.State,
+					ToSignature: sig,
+				})
+				if err != nil {
+					log.Debug(err)
+					continue
+				}
+				if err := rm(d, e.State.Id.Id); err != nil {
+					log.Debug(err)
+				}
+			}
+			return nil
```

Comment:
why never return any error?

Reply:
i think any error can be ignored when batch close persisted channel_state, because ledger service will make sure all failed transactions be rollback eventually.
i tried to cache all errors and return them finally but i failed for the `WithContext` function return param.
can be fixed by creating client every loop if needed.

Reply:
At least one error should be thrown after the full `for _, e := range states` is done. Otherwise this function will never correctly throw err back up. It is the parent's responsibility to ignore/log/do whatever.

Reply:
Logically it is the same, but throwing error makes this function `CloseLedgerChannel` more consistent and can be unit tested when needed.

Reply:
just return the last error is ok.

Reply:
make sense